### PR TITLE
Add cost tracking per LLM to proxy

### PR DESCRIPTION
**Pull request by ishaan-jaff**
Problem: 
It's hard to track how much I have spent when running my proxy. I need to see costs incurred per model on the proxy 

Proposed solution V0:
use `litellm.completion_cost` to track costs per llm 

for each request write costs to costs.json (in a separate thread) [Non-blocking]

## Example costs.json

Screenshot 2023-09-09 at 6 42 30 PM

Next steps:
- If this looks we can add better tracking for costs + even make a UI dashboard 
Example below:
Screenshot 2023-07-28 at 5 36 19 PM

### CODE REVIEW

File: proxy/cost.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (ishaan-jaff) [inline proxy/cost.json:1]: this is how costs will be tracked per model for v0

Comment (yujonglee) [reply proxy/cost.json:1]: Result looks awesome!

BTW, let's remove this file since it is not part of the code.

**Review by yujonglee — CHANGES_REQUESTED** [review]
@ishaan-jaff 
Great!

Since `user specific budget` will be added in #4, I think we can slightly modify this to PR to do the following.

1. Remove JSON writing
2. Instead, expose report/reset and report/current (feel free to suggest alternative names).
3. When I send a request to report/reset, the report will be reset (currently, the report only contains the API cost).
4. When I send a request to report/current, I will receive the current report, which includes the cost.

My use-case is to get how much $ is spent for a each evaluation run.

**Comment by yujonglee** [comment]
@ishaan-jaff  Please see https://github.com/fastrepl/proxy/pull/6/commits/92db8de10de0ac0801de4f4a60c7ca7be8e86b12 and share your thought. Only functional difference is remove file read/write. Let me know if you think it is needed.

**Pull request closed without merging**